File: backend/jobplatform/src/main/java/edu/tlu/jobplatform/cv/domain/model/CVTemplate.java
package edu.tlu.jobplatform.cv.domain.model;

import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * Aggregate -- template CV do admin quan ly qua API.
 *
 * htmlContent luu toan bo XHTML hop le trong DB.
 * Khong con file classpath -- moi template deu qua Admin API.
 */
@Getter
@Builder
public class CVTemplate {

    private final UUID id;
    private String name;
    private String thumbnailUrl;
    private String category; // "professional" | "creative" | "simple"
    private boolean premium;
    private String htmlContent; // XHTML string day du, luu trong DB
    private boolean active; // false = an khoi danh sach candidate
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    // Business rules

    public void update(String name, String thumbnailUrl, String category,
            boolean premium, String htmlContent, boolean active) {
        this.name = name;
        this.thumbnailUrl = thumbnailUrl;
        this.category = category;
        this.premium = premium;
        this.htmlContent = htmlContent;
        this.active = active;
    }

    public void activate() {
        this.active = true;
    }

    public void deactivate() {
        this.active = false;
    }
}

File: frontend/src/domain/models/CVTemplate.ts
// ─────────
// Domain Model: CVTemplate
// Maps 1-1 với CVTemplateJpaEntity / CVTemplate.java ở backend
// ─────────

export type TemplateCategory = "professional" | "creative" | "simple";

export interface CVTemplate {
  id: string;
  name: string;
  thumbnailUrl: string | null;
  /** "professional" | "creative" | "simple" */
  category: TemplateCategory | null;
  /** Chỉ dành cho tài khoản premium */
  premium: boolean;
  /** Ứng viên có thể nhìn thấy và chọn */
  active: boolean;
  /** Thymeleaf HTML — chỉ có khi gọi detail */
  htmlContent: string | null;
  createdAt: string | null;
  updatedAt: string | null;
}

// ── Value objects dùng khi tạo / cập nhật ─────

export interface CreateTemplateData {
  name: string;
  thumbnailUrl: string;
  category: TemplateCategory;
  premium: boolean;
  /** Nội dung HTML/Thymeleaf đầy đủ */
  htmlContent: string;
}

export interface UpdateTemplateData extends CreateTemplateData {
  active: boolean;
}